modules/references/tests/server/reference.server.jobs.tests.js

```diff
@@ -0,0 +1,115 @@
+'use strict';
+
+var async = require('async'),
+    crypto = require('crypto'),
+    moment = require('moment'),
+    mongoose = require('mongoose'),
+    path = require('path'),
+    should = require('should'),
+    sinon = require('sinon'),
+    config = require(path.resolve('./config/config')),
+    jobPublishReference = require('../../server/jobs/references-publish.server.job'),
+    Reference = mongoose.model('Reference');
+
+describe('Job: Set reference to public after a given period of time', function () {
+
+  // fake Date
+  // stub config.limits.timeToReplyReference with custom test value
+  beforeEach(function () {
+    sinon.useFakeTimers({ now: new Date('2018-10-12 11:33:21.312'), toFake: ['Date'] });
+    sinon.stub(config.limits, 'timeToReplyReference').value({ days: 7 });
+  });
+
+  afterEach(function () {
+    sinon.restore();
+  });
+
+  // delete references after each test
+  afterEach(function (done) {
+    Reference.deleteMany({}).exec(done);
+  });
+
+  function generateRandomId() {
+    var result = crypto.randomBytes(12).toString('hex');
+    // padding with 0 to 24 characters
+    result = '0'.repeat(24 - result.length) + result;
+    return result;
+  }
+
+  function generateRandomBoolean() {
+    return !!Math.floor(2 * Math.random());
+  }
+
+  function generateRandomRecommend() {
+    return ['yes', 'no', 'unknown'][Math.floor(3 * Math.random())];
+  }
+
+  var referenceData = {
+    get userFrom() { return generateRandomId(); },
+    get userTo() { return generateRandomId(); },
+    // here we don't care if all interactions are false
+    get met() { return generateRandomBoolean(); },
+    get hostedMe() { return generateRandomBoolean(); },
+    get hostedThem() { return generateRandomBoolean(); },
+    get recommend() { return generateRandomRecommend(); }
```

Comment:
This caught my eye as it looked _fancy_ to use object getters.

I can imagine this causing some confusion as repeated calls (e.g. `referenceData.userFrom`) will keep returning different values.

I would find something like this less confusing:
```js
function generateReferenceData () {
  return {
    userFrom: generateRandomId(),
    // ... etc
  }
}
```

Reply:
Agreed here; these were simplified in https://github.com/Trustroots/trustroots/pull/723/commits/3c523d1a42fabde58aa07fdf39674ed29f66ebf8